shopelectro/views/catalog.py

```diff
@@ -171,24 +228,43 @@ class CategoryPage(catalog.CategoryPage):
 
     def get_context_data(self, **kwargs):
         """Add sorting options and view_types in context."""
-        context_ = (
-            context.Category(
-                url_kwargs=self.kwargs,
-                request=self.request,
-                page=self.object,
-                products=models.Product.objects.all(),
-                product_pages=models.ProductPage.objects.all(),  # Ignore CPDBear
-            )
-            | context.TaggedCategory(tags=models.Tag.objects.all())
-            | context.SortingCategory()  # requires TaggedCategory
-            | context.PaginationCategory()  # requires SortingCategory
-            | context.ProductBrands()  # requires TaggedCategory
-            | se_context.ProductImages()
-            | context.DBTemplate()  # requires TaggedCategory
+        sorting_index = int(self.kwargs.get('sorting', 0))
+
+        # @todo #683:30m Create Page context class.
+        #  Move rest of plain context data to this class.
```

Comment:
it's not clear task for me. I don't understand what is `rest of plain context data`.

And `get_catalog_context` big separated method not looks good. This task request to remove it in favour of some class as i suggest.
If my suggestion is right, i agree with this thought. Context should receive this set of args in more clear way, than it does now

Reply:
I will clarify it